lib/install.js

```diff
@@ -120,6 +120,11 @@ function handleResolvedPkg (dir, pkg, cb) {
       // package level.
       fs.rename.bind(null, tmpDest, finalDest)
     ], function (err) {
+      // Include package.json's metadata.
+      // `pkg` at this point just comes from the resolver, which only has
+      // { name, version, uid, shasum, tarball }.
+      pkg = assign({}, require(path.join(dir, pkg.uid, 'package.json')), pkg)
```

Comment:
btw, it seems my implementation only performs binstub linking on top-level dependencies... I _think_ this is consistent with npm behavior. doing it at this level would install binstubs for deep dependencies, wouldn't it?